src/tests/ftest/container/fill_destroy_loop.py

```diff
@@ -0,0 +1,97 @@
+"""
+  (C) Copyright 2022 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+import random
+
+from apricot import TestWithServers
+from general_utils import get_random_bytes, DaosTestError
+from test_utils_container import TestContainerData
+
+
+class BoundaryPoolContainerSpace(TestWithServers):
+    """Test class for pool Boundary testing.
+
+    Test Class Description:
+        Test to create a pool and container and write random data to fill the container and delete
+        container, repeat the test 100 times as boundary condition.
+
+    :avocado: recursive
+    """
+
+    DER_NOSPACE = "-1007"
+
+    def write_pool_until_nospace(self):
+        """write pool and container until pool is full. """
+        data_size_base = self.params.get("data_size_base", "/run/container/*")
+        container = self.get_container(self.pool)
+        container.open()
+        written_byte = 0
+        write_count = 1
+        while (1):
+            data_size = random.randint(data_size_base * 0.5, data_size_base * 1.5) # nosec
+            written_byte += data_size
+            try:
+                container.written_data.append(TestContainerData(False))
+                container.written_data[-1].write_record(
+                    container,
+                    get_random_bytes(container.akey_size.value),
+                    get_random_bytes(container.dkey_size.value),
+                    get_random_bytes(data_size))
+                self.log.info("--{0} wrote container-obj, sz {1}".format(write_count, data_size))
```

Comment:
(pylint-logging-format-interpolation) Use lazy % formatting in logging functions